cache_path.go

```diff
@@ -56,25 +57,51 @@ func parseIgnoreList(list []string) map[string]bool {
 	return ignoreByPath
 }
 
+func isSymlink(pth string) (bool, error) {
+	linkFileInfo, err := os.Lstat(pth)
+	if err != nil {
+		return false, fmt.Errorf("failed to get file info, error: %s", err)
+	}
+
+	return linkFileInfo.Mode()&os.ModeSymlink != 0, nil
+}
+
 // expandPath returns every file included in pth (recursively) if it is a dir,
 // if pth is a file it will be returned as an array.
-func expandPath(pth string) ([]string, error) {
-	var subPaths []string
-	if err := filepath.Walk(pth, func(p string, i os.FileInfo, err error) error {
+func expandPath(root string) (regularFiles []string, irregularPaths []string, err error) {
+	if err := filepath.Walk(root, func(path string, i os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Ignoring symlink target changes for cache invalidation.
+		// If case it a link to a directory outside of the cached paths (e.g. yarn workspaces),
+		// will not add the linked directory to the cache, and will not invalidate cache if it changes.
```

Comment:
// will not add the linked directory to the cache, and will not invalidate the cache if it changes, as we expect them to be part of the repository.